os/board/rtl8721csm/src/component/common/file_system/ftl/ftl.c

```diff
@@ -1013,9 +1067,41 @@ uint32_t ftl_read(uint16_t logical_addr, uint32_t *value)
     return  ret;
 }
 
+#if defined(CONFIG_AMEBAD_TRUSTZONE) && (FTL_SECURE == 1)
+/* FTL Secure Save and Load API */
+uint32_t ftl_secure_save_to_storage(void *pdata_tmp, uint16_t offset, uint16_t size)
+{
+	uint32_t ret = 0;
+	uint8_t* tmp_buff = NULL;
+
+	tmp_buff = rtw_zmalloc(size);
+	if (tmp_buff == NULL) {
+		return 0x10;
+	}
+
+	up_allocate_secure_context(FTL_SECURE_CONTEXT_SIZE);
+	ret = ameba_ftl_save_to_storage(tmp_buff, pdata_tmp, offset, size);

```

Comment:
Can we check for return value here and handle error scenarios ?

Reply:
The error scenarios will be handle on the caller side.
We can print error log here is want to.

Reply:
yes, I think that would be good for debugging purpose

Reply:
Added log for error scenarios, return value will be print if error occurs.